crates/routes/src/feeds.rs

```diff
@@ -520,13 +521,28 @@ fn create_post_items(
     p.counts.comments);
 
     // If its a url post, add it to the description
-    let link = Some(if let Some(url) = p.post.url {
+    // and see if we can parse it as a media enclosure
+    let mut enclosure_opt = None;
+    if let Some(url) = p.post.url {
       let link_html = format!("<br><a href=\"{url}\">{url}</a>");
       description.push_str(&link_html);
-      url.to_string()
-    } else {
-      post_url.clone()
-    });
+
+      // Guess MIME media type from url, or fail silently by returning None
+      enclosure_opt = Url::parse(url.as_str())
+        .ok()
+        .and_then(|u| {
+          u.path_segments()
+            .and_then(|s| s.last().map(|s| s.to_string()))
+        })
+        .map(|f| {
+          let mime = mime_guess::from_path(&f).first_or_octet_stream();
+          let mut enclosure = Enclosure::default();
+          enclosure.set_url(url.as_str());
+          enclosure.set_mime_type(mime.to_string());
+          enclosure.set_length("0".to_string());
+          enclosure
+        });
+    }
```

Comment:
Use `post.url_content_type` instead which was recently added.

Reply:
Thanks for the feedback, I switched to `post.url_content_type` but left the `0` as its from the specification:

> When an enclosure's size cannot be determined, a publisher should use a length of 0.

https://www.rssboard.org/rss-profile#element-channel-item-enclosure

Reply:
Its still using mime_guess, did you forget to push your changes?

Reply:
Ah sorry I misunderstood you, I just pushed a commit to make proper use of it, let me know what you think.